DOC: Fix assume_centered parameter documentation in EmpiricalCovariance (#31809)
scikit-learn/scikit-learn: doc/modules/covariance.rst

:class:`EmpiricalCovariance` object to the data sample with the
:meth:`EmpiricalCovariance.fit` method. Be careful that results depend
on whether the data are centered, so one may want to use the

> ``assume_centered`` parameter accurately. More precisely, if 
> ``assume_centered=True``, then all features in `X` should have a mean of zero. 
> If not, the data should be centered by the user, or `assume_centered=False` should be used.

Reviewer: Let's keep a reference to the train and test set to remind that both should be centered
```suggestion
`assume_centered` parameter accurately. More precisely, if `assume_centered=True`, then
all features in the train and test sets should have a mean of zero. If not, both should
be centered by the user, or `assume_centered=False` should be used.
```
